torch/_inductor/ir.py

```diff
@@ -4654,18 +4657,37 @@ def __init__(self, *, kernel_idx, grid, kernel_args):
         self.kernel_idx = kernel_idx
         self.grid = grid
 
-        kernel, _ = self.get_kernel_and_configs()
+        kernel, configs = self.get_kernel_and_configs()
         # If we are autotuning, not all arguments will be passed
         self.ordered_kwargs_for_cpp_kernel = [
             arg for arg in kernel.arg_names if arg in kernel_args
         ]
 
-        mark_node_as_mutating(
-            self, *[a for a in kernel_args.values() if isinstance(a, TensorBox)]
-        )
+
+        from torch._higher_order_ops.triton_kernel_wrap import identify_mutated_tensors
+
+        def _to_real_values(v):
```

Comment:
we already do this type of conversation in identify_mutated_tensors. From FakeTensor to real tensor, why dont you just add TensorBox there?

Reply:
yeah not sure why I was afraid to add it there